core/services/mongodb-local/src/main/java/com/mongodb/stitch/core/services/mongodb/local/internal/EmbeddedMongoClientFactory.java

```diff
@@ -17,57 +17,78 @@
 package com.mongodb.stitch.core.services.mongodb.local.internal;
 
 import com.mongodb.client.MongoClient;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 import org.bson.codecs.configuration.CodecRegistry;
 
 public abstract class EmbeddedMongoClientFactory {
-
```

Comment:
Im not super familiar with how this code used to work, but it seems strange that we are no longer caring about storing other MongoClient instances. Are we just saying that you should only have one at a time?

Reply:
The previous code, in practice, only stored a maximum of one key. This is because in current state, we can only have one active client at any given time and the rest must be aggressively closed.

Reply:
Got it